Mage/Archivist/Archivist.cs

```diff
@@ -133,12 +133,18 @@ public class Archivist : EventEmitter<JObject> {
 			//if (result == null && !options ["optional"]) {
 			//	return cb (new Exception ("ValueNotFound"), null);
 			//}
-			
-			// Create vault value
-			VaultValue newValue = new VaultValue((JObject)result);
-			
-			// Add value to cache
-			_cache.Add (cacheKeyName, newValue);
+
+			VaultValue newValue;
+			try {
+				// Create vault value
+				newValue = new VaultValue((JObject)result);
+				
+				// Add value to cache
+				_cache.Add (cacheKeyName, newValue);
+			} catch (Exception cacheError) {
+				cb(cacheError, null);
```

Comment:
Any chance these callbacks could have `null` as a default value to the 2nd argument? Minor detail, and perhaps you guys like it to be very explicit (in which case I won't argue), but in the country I come from (JS-land) we don't pass extra arguments after passing an error to a callback.